src/flags/run_time_configurable.cpp

```diff
@@ -10,102 +10,170 @@
-  *(bolt_server_name_.Lock()) = *name;
-  // Override server name if passed via command line argument
-  if (!FLAGS_bolt_server_name_for_init.empty()) {
-    memgraph::utils::global_settings.SetValue(kServerNameSettingKey, FLAGS_bolt_server_name_for_init);
+auto ToLLEnum(std::string_view val) {
+  const auto ll_enum = memgraph::flags::LogLevelToEnum(val);
+  if (!ll_enum) {
+    throw memgraph::utils::BasicException("Unsupported log level {}", val);
   }
+  return *ll_enum;
+}
 
-  // Register query timeout
-  memgraph::utils::global_settings.RegisterSetting(kQueryTxSettingKey, kDefaultQueryTx, [&] {
-    const auto query_tx = memgraph::utils::global_settings.GetValue(kQueryTxSettingKey);
-    MG_ASSERT(query_tx, "Query timeout is missing from the settings");
-    execution_timeout_sec_ = std::stod(*query_tx);
-  });
-  // Update value from read settings
-  const auto &tx = memgraph::utils::global_settings.GetValue(kQueryTxSettingKey);
-  MG_ASSERT(tx, "Failed to read query timeout from settings.");
-  execution_timeout_sec_ = std::stod(*tx);
-  // Override query timeout if passed via command line argument
-  if (FLAGS_query_execution_timeout_sec != -1) {
-    memgraph::utils::global_settings.SetValue(kQueryTxSettingKey, std::to_string(FLAGS_query_execution_timeout_sec));
-  }
+bool ValidBoolStr(std::string_view in) {
+  const auto lc = memgraph::utils::ToLowerCase(in);
+  return lc == "false" || lc == "true";
+}
 
-  // Register log level
-  auto get_global_log_level = []() {
-    const auto log_level = memgraph::utils::global_settings.GetValue(kLogLevelSettingKey);
-    MG_ASSERT(log_level, "Log level is missing from the settings");
-    const auto ll_enum = memgraph::flags::LogLevelToEnum(*log_level);
-    if (!ll_enum) {
-      throw utils::BasicException("Unsupported log level {}", *log_level);
+auto GenHandler(std::string flag, std::string key) {
+  return [key = std::move(key), flag = std::move(flag)]() -> std::string {
+    const auto &val = memgraph::utils::global_settings.GetValue(key);
+    MG_ASSERT(val, "Failed to read value at '{}' from settings.", key);
+    gflags::SetCommandLineOption(flag.c_str(), val->c_str());
+    return *val;
+  };
+}
+
+}  // namespace
+
+namespace memgraph::flags::run_time {
+
+void Initialize() {
+  constexpr bool kRestore = true;
+  auto register_flag = [&](
+                           const std::string &flag, const std::string &key, bool restore,
+                           std::function<void(const std::string &)> post_update = [](auto) {},
+                           std::function<bool(std::string_view)> validator = [](std::string_view) { return true; }) {
+    // Get flag info
+    gflags::CommandLineFlagInfo info;
+    gflags::GetCommandLineFlagInfo(flag.c_str(), &info);
+    // Register setting
+    auto update = GenHandler(flag, key);
+    memgraph::utils::global_settings.RegisterSetting(
+        key, info.default_value,
+        [update, post_update] {
```

Comment:
I think you can capture by ref here
```suggestion
        [&update, &post_update] {
```